Create documentation for bundles docs (#3804)
kyma-project/kyma: docs/helm-broker/docs/03-04-bundles-docs.md

---
title: Bundles docs
type: Details
---

## Documentation for ServiceClasses provided by bundle

The Helm Broker in the provisioning process pushes the `addonRepositoryURL` variable into installed chart. With this feature a bundle which provides its own ServiceClasses e.g. `Azure Broker` can also install a documentation for them.

If the bundle provides ServiceClasses and you want to have a documentation for it, you should create a `docs.yaml` inside the bundle's chart with the following structure:
```
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  labels:
    cms.kyma-project.io/view-context: service-catalog
  name: {ServiceClass ID}
spec:
  displayName: "Slack Connector Add-on"
  description: "Overall documentation, OpenAPI and AsyncAPI for Slack Connector Add-on"
  sources:
    - type: {type}
      name: {name}  # must be unique within sources
      mode: {mode}
      url: {.Values.addonsRepositoryURL}
      filter: docs/{class_name}/   # provide filter only if you use the package mode
___

> # it is prefered to store all of the ClusterDocsTopics definitions in one file

Reviewer: this sentence: "It is preferred" should be moved from the example to the normal documentation paragraph

Author: done

---
title: Bundles docs
type: Details
---

## Documentation for ServiceClasses provided by bundle

The Helm Broker in the provisioning process pushes the `addonRepositoryURL` variable into installed chart. With this feature a bundle which provides its own ServiceClasses e.g. `Azure Broker` can also install a documentation for them.

If the bundle provides ServiceClasses and you want to have a documentation for it, you should create a `docs.yaml` inside the bundle's chart with the following structure:
```
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  labels:
    cms.kyma-project.io/view-context: service-catalog
  name: {ServiceClass ID}
spec:

>   displayName: "Slack Connector Add-on"

Reviewer: Should we use placeholder rather than examples?

Author: @klaudiagrz What do you think?

Reviewer 2: @klaudiagrz ? :)

  description: "Overall documentation, OpenAPI and AsyncAPI for Slack Connector Add-on"
  sources:
    - type: {type}
      name: {name}  # must be unique within sources
      mode: {mode}
      url: {.Values.addonsRepositoryURL}
      filter: docs/{class_name}/   # provide filter only if you use the package mode
___
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata ... 
```

> 

Reviewer 2: BUT! ;) I'm not sure if we should mention it at all if everything is described here: https://github.com/kyma-project/kyma/blob/master/docs/headless-cms/docs/06-02-clusterdocstopic.md.
Perhaps you could just make a reference to this doc and say sth like: `For detailed descriptions of all parameters, see the [ClusterDocsTopic custom resource](/components/headless-cms/#custom-resource-clusterdocstopic).` Hm?

> 

Author: I already link that doc in the 03-01-create-bundles doc so I will leave like it is. Also, for now, the fields are the same but we can add there some new fields so it's better to leave that column.

> 

Reviewer 3: Actually, I agree with @kazydek. Instead of describing the fields here, make a reference to the relevant documentation. We don't know if and when we're going to add more fields.

> One ClusterDocsTopic object corresponds to a single ServiceClass with the same ID as the name of specified ClusterDocsTopic.

Reviewer 2: ```suggestion
One ClusterDocsTopic object corresponds to a single Service Class with the same ID as the name of the specified ClusterDocsTopic.
```

Reviewer 2: meaning...? :)

Author: discussed

> You should hold the ClusterDocsTopics definitions for a given broker in one file.

Reviewer 2: ```suggestion
Store the ClusterDocsTopics definitions for a given Service Broker in one file.
```

> If you want to provide documentation for every class provided by the broker you must also provide the ClusterDocsTopics for all of them.

Reviewer 2: ```suggestion
If you want to deliver documentation for every Service Class provided by a given Service Broker, you must create ClusterDocsTopics for all of them.
```

> The documentation for the ServiceClasses provided by a bundle should be stored in the `docs/{class_name}` folder.

Reviewer 2: ```suggestion
Store the documentation for the Service Classes provided by a bundle in a `docs/{class_name}` folder.
```

```
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  labels:
    cms.kyma-project.io/view-context: service-catalog
  name: {ServiceClass ID}
spec:
  displayName: {displayName}
  description: {description}
  sources:
    - type: {type}
      name: {name}
      mode: {mode}
      url: {.Values.addonsRepositoryURL}
      filter: docs/{class_name}/ 
___
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata ... 
```
For detailed descriptions of all parameters, see the [ClusterDocsTopic custom resource](/components/headless-cms/#custom-resource-clusterdocstopic). 
For more information about currently supported types of the assets, read [this](/components/headless-cms/#overview-overview-headless-cms-in-kyma) document.

One ClusterDocsTopic object corresponds to a single Service Class with the same ID as the name of the specified ClusterDocsTopic.
Place all ClusterDocsTopic definitions for your bundle in the `docs.yaml` file.
Store documentation for each Service Class in the `docs/{class_name}` directory which corresponds to the value of the **filter** parameter in the ClusterDocsTopic definition.

> During the provisioning process, the Helm Broker pushes the **addonRepositoryURL** variable into the chart. The **addonsRepositoryURL** points to your bundle compressed to a `.tgz` file.

Reviewer 3: I'm not sure about this paragraph. To be honest, I would totally omit this info. @PK85  let us know what you think :)
